include/kernels/dslash_shmem_helper.cuh

```diff
@@ -51,4 +51,39 @@ namespace quda {
     __device__ void operator()(int i) { *(arg.arr + i) = arg.val; }
   };
 
+#ifdef NVSHMEM_COMMS
+  /**
+       @brief Parameter structure for driving init_dslash_arr
+    */
+  template <typename T_> struct shmem_signal_wait_arg : kernel_param<> {
+    using T = T_;
+    T *sync_arr;
+    T counter;
+    int commDim[8];
+
+    shmem_signal_wait_arg() :
+      kernel_param(dim3(8, 1, 1)),
+      sync_arr(dslash::get_exchangeghost_shmem_sync_arr()),
+      counter((activeTuning() && !policyTuning()) ? 2 : dslash::get_exchangeghost_shmem_sync_counter())
+    {
+      for (int d = 0; d < 4; d++) {
+        commDim[2 * d] = comm_dim_partitioned(d);
+        commDim[2 * d + 1] = comm_dim_partitioned(d);
+      }
+    }
+  };
+
+  /**
+     @brief Functor to initialize the arrive signal
```

Comment:
I think this comment is a bit misleading: doesn't this functor do the wait for the signals to arrive, rather than initializing the arrival signal?

Reply:
yep, looks like copy/paste gone wrong